x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/utils/import_rules_utils.ts

```diff
@@ -0,0 +1,396 @@
+                    rule,
+                  });
+                  await patchRules({
+                    rulesClient,
+                    savedObjectsClient,
+                    author,
+                    buildingBlockType,
+                    spaceId,
+                    ruleStatusClient,
+                    description,
+                    enabled,
+                    eventCategoryOverride,
+                    falsePositives,
+                    from,
+                    query,
+                    language,
+                    license,
+                    outputIndex,
+                    savedId,
+                    timelineId,
+                    timelineTitle,
+                    meta,
+                    filters,
+                    rule: migratedRule,
+                    index,
+                    interval,
+                    maxSignals,
+                    riskScore,
+                    riskScoreMapping,
+                    ruleNameOverride,
+                    name,
+                    severity,
+                    severityMapping,
+                    tags,
+                    timestampOverride,
+                    throttle,
+                    to,
+                    type,
+                    threat,
+                    threshold,
+                    threatFilters,
+                    threatIndex,
+                    threatIndicatorPath,
+                    threatQuery,
+                    threatMapping,
+                    threatLanguage,
+                    concurrentSearches,
+                    itemsPerSearch,
+                    references,
+                    note,
+                    version,
+                    exceptionsList: [...exceptions],
+                    anomalyThreshold,
+                    machineLearningJobId,
+                    actions,
+                  });
+                  resolve({
+                    rule_id: ruleId,
+                    status_code: 200,
+                  });
+                } else if (rule != null) {
+                  resolve(
+                    createBulkErrorObject({
+                      ruleId,
+                      statusCode: 409,
+                      message: `rule_id: "${ruleId}" already exists`,
+                    })
+                  );
+                }
+              } catch (err) {
+                resolve(
+                  createBulkErrorObject({
+                    ruleId,
+                    statusCode: err.statusCode ?? 400,
+                    message: err.message,
+                  })
+                );
+              }
+            } catch (error) {
+              reject(error);
+            }
+          });
+          return [...accum, importsWorkerPromise];
+        }, [])
+      );
+      importRuleResponse = [...importRuleResponse, ...newImportRuleResponse];
+    }
+
+    return importRuleResponse;
+  }
+};
+
+// TODO: Batch this upfront and send down to check against
```

Comment:
I would assume you want to keep this TODO here, but just pointing it out in case you wanted to change it before merging.

Reply:
Yea, gonna do this as follow up. There's a few performance related enhancements I'd like to follow through on.